Compiling C with Zig 0.8.0 fails on macOS Mojave

We're currently trying to update Homebrew's Zig to 0.8.0. This builds fine, but the test fails at building a `Hello, world!` program in C, but only on Mojave (macOS 10.14). Everything works fine on Catalina (10.15) and Big Sur (11, both Intel and ARM).

It returns a number of errors of the form:
```
==> /usr/local/Cellar/zig/0.8.0/bin/zig cc hello.c -o hello
In file included from hello.c:1:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/stdio.h:64:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/_stdio.h:93:16: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
        unsigned char   *_base;
                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/_stdio.h:93:16: note: insert '_Nullable' if the pointer may be null
        unsigned char   *_base;
                        ^
                          _Nullable 
```

We build Zig with the following CMake invocation:
```
cmake . -DZIG_STATIC_LLVM=ON -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
```

The version bump PR is https://github.com/Homebrew/homebrew-core/pull/78774, and CI logs are available at https://github.com/Homebrew/homebrew-core/pull/78774/checks?check_run_id=2753820571. The failures start at https://github.com/Homebrew/homebrew-core/pull/78774/checks?check_run_id=2753820571#step:7:55.

Assistance on this would be appreciated.

Thanks for the report @carlocab. The error is actually the below:

```
In file included from hello.c:1:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/stdio.h:224:5: error: 'TARGET_OS_EMBEDDED' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_EMBEDDED
```

--- a/src/main.zig
+++ b/src/main.zig
@@ -1664,8 +1664,8 @@ fn buildOutputType(
 
         const has_sysroot = if (comptime std.Target.current.isDarwin()) outer: {
             const min = target_info.target.os.getVersionRange().semver.min;
-            const at_least_catalina = min.major >= 11 or (min.major >= 10 and min.minor >= 15);
-            if (at_least_catalina) {
+            const at_least_mojave = min.major >= 11 or (min.major >= 10 and min.minor >= 14);
+            if (at_least_mojave) {
                 const sdk_path = try std.zig.system.getSDKPath(arena);
                 try clang_argv.ensureCapacity(clang_argv.items.len + 2);
                 clang_argv.appendAssumeCapacity("-isysroot");

